From 75eb3ff25ac9b36a66dbf6d7f9858981cd57aff9 Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Sat, 3 Jan 2026 15:55:57 +0800
Subject: [PATCH 63/63] loonggpu: handle the pci_resize_resource API change in
 Linux 6.19

The loonggpu_device_resize_fb_bar() function seems not used as at now so
I'm unsure if the change is correct (see TODO).

Link: https://git.kernel.org/torvalds/c/db92e3fef53e
Link: https://git.kernel.org/torvalds/c/337b1b566db0
Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 loonggpu/loonggpu_device.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/loonggpu/loonggpu_device.c b/loonggpu/loonggpu_device.c
index d480ac6..90917dc 100644
--- a/loonggpu/loonggpu_device.c
+++ b/loonggpu/loonggpu_device.c
@@ -446,11 +446,16 @@ int loonggpu_device_resize_fb_bar(struct loonggpu_device *adev)
 	pci_write_config_word(adev->pdev, PCI_COMMAND,
 			      cmd & ~PCI_COMMAND_MEMORY);
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
+	/* TODO: do we need to preserve some BAR(s) via the 4th parameter?  */
+	r = pci_resize_resource(adev->pdev, 0, rbar_size, 0);
+#else
 	pci_release_resource(adev->pdev, 3);
 
 	pci_release_resource(adev->pdev, 0);
 
 	r = pci_resize_resource(adev->pdev, 0, rbar_size);
+#endif
 	if (r == -ENOSPC)
 		DRM_INFO("Not enough PCI address space for a large BAR.");
 	else if (r && r != -ENOTSUPP)
-- 
2.52.0

